modules/SofaGeneralLoader/src/SofaGeneralLoader/ReadState.inl

```diff
@@ -230,8 +237,16 @@ void ReadState::processReadState()
         str >> cmd;
         if (cmd == "X=")
         {
-            mmodel->readVec(core::VecId::position(), str);            
-            mmodel->applyScale(d_scalePos.getValue(), d_scalePos.getValue(), d_scalePos.getValue());
+            mmodel->readVec(core::VecId::position(), str);
+
+            if (d_scalePos.getValue() != 1.)
```

Comment:
1- Testing an equality of float-point numbers is never a good idea (actually I am doubtful that these tests are even really necessary)
3- maybe store the scalePos/rotation/translation value once in a const variable and use those later (instead of calling everytime getValue())

Reply:
Sure. This is just a copy past of what was done in MechanicalObject. I guess I can make the changes in both files. 
1- I guess the idea was to avoid unnecessary computations, but I don't mind removing them.